python/ray/data/read_api.py

```diff
@@ -132,11 +137,21 @@ def from_items(items: List[Any], *, parallelism: int = -1) -> Dataset[Any]:
     metadata: List[BlockMetadata] = []
     for i in builtins.range(detected_parallelism):
         stats = BlockExecStats.builder()
-        builder = DelegatingBlockBuilder()
+        if output_arrow_format:
+            builder = ArrowBlockBuilder()
+        else:
+            builder = SimpleBlockBuilder()
```

Comment:
shouldn't we do `builder = DelegatingBlockBuilder()` here? We shouldn't change default behavior, right?

Reply:
This defeats the purpose right? We still have the implicit arrow block to simple block fallback even if output_arrow_format is False

Reply:
reverted back to delegatingblockbuilder